File: memento/src/main/java/com/alexstyl/specialdates/events/namedays/NamedayLocale.java
package com.alexstyl.specialdates.events.namedays;

public enum NamedayLocale {
    GREEK("gr", true),
    ROMANIAN("ro", false),
    RUSSIAN("ru", false),
    LATVIAN("lv", false),
    LATVIAN_EXTENDED("lv_ext", false),
    SLOVAK("sk", false),
    ITALIAN("it", false),
    CZECH("cs", false),
    HUNGARIAN("hu", false);

    private final String shortCode;
    private final boolean soundCompared;

    NamedayLocale(String shortCode, boolean soundCompared) {
        this.shortCode = shortCode;
        this.soundCompared = soundCompared;
    }

    public static NamedayLocale from(String displayLanguage) {
        for (NamedayLocale locale : values()) {
            if (locale.getCountryCode().equalsIgnoreCase(displayLanguage)) {
                return locale;
            }
        }
        throw new IllegalArgumentException("No NamedayLocale found for "+ displayLanguage);
    }

    public boolean isComparedBySound() {
        return soundCompared;
    }

    public String getCountryCode() {
        return shortCode;
    }
}

File: memento/src/main/java/com/alexstyl/specialdates/events/namedays/NamedayLocale.kt
package com.alexstyl.specialdates.events.namedays

enum class NamedayLocale(val countryCode: String) {
    GREEK("gr"),
    ROMANIAN("ro"),
    RUSSIAN("ru"),
    LATVIAN("lv"),
    LATVIAN_EXTENDED("lv_ext"),
    SLOVAK("sk"),
    ITALIAN("it"),
    CZECH("cs"),
    HUNGARIAN("hu");


    companion object {

        fun from(displayLanguage: String): NamedayLocale {
            for (locale in values()) {
                if (locale.countryCode.equals(displayLanguage, ignoreCase = true)) {
                    return locale
                }
            }
            throw IllegalArgumentException("No NamedayLocale found for $displayLanguage")
        }
    }
}
